azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/boot/ApplicationInsightsModuleConfiguration.java

```diff
@@ -0,0 +1,125 @@
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.core.env.Environment;
+
+/**
+ * {@link Configuration} for non-web applications.
+ *
+ * @author Arthur Gavlyukovskiy, Dhaval Doshi
+ */
+@Configuration
+@EnableConfigurationProperties(ApplicationInsightsProperties.class)
+@ConditionalOnProperty(value = "azure.application-insights.enabled", havingValue = "true", matchIfMissing = true)
+public class ApplicationInsightsModuleConfiguration {
+
+    private ApplicationInsightsProperties applicationInsightsProperties;
+
+    @Autowired
+    public ApplicationInsightsModuleConfiguration(ApplicationInsightsProperties properties) {
+        this.applicationInsightsProperties = properties;
+    }
+
+    @Bean
+    public SdkVersionContextInitializer sdkVersionContextInitializer() {
+        return new SdkVersionContextInitializer();
+    }
+
+    @Bean
+    public DeviceInfoContextInitializer deviceInfoContextInitializer() {
+        return new DeviceInfoContextInitializer();
+    }
+
+    @Bean
+    public SpringBootTelemetryInitializer springBootTelemetryInitializer() {
+        return new SpringBootTelemetryInitializer();
+    }
+
+    //FIXME: This should be conditional on operating System. However, current architecture of ProcessBuiltInPerformanceCountersFactory
+    //FIXME: does not separate this concerns therefore cannot condition as of now.
+    @Bean
+    @DependsOn("performanceCounterContainer")
+    @ConditionalOnProperty(value = "azure.application-insights.default-modules.ProcessPerformanceCountersModule.enabled", havingValue = "true", matchIfMissing = true)
+    public ProcessPerformanceCountersModule processPerformanceCountersModule() {
+        try {
+            return new ProcessPerformanceCountersModule();
+        }
+        catch (Exception e) {
+            throw new IllegalStateException("Could not initialize Windows performance counters module, " +
+                    "please set property 'azure.application-insights.default-modules.ProcessPerformanceCountersModule.enabled=false' to avoid this error message.", e);
+        }
+    }
+
+    @Bean
+    @DependsOn("performanceCounterContainer")
+    @ConditionalOnProperty(value = "azure.application-insights.default.modules.JvmPerformanceCountersModule.enabled", havingValue = "true", matchIfMissing = true)
+    public JvmPerformanceCountersModule jvmPerformanceCountersModule() {
+        try {
+            return new JvmPerformanceCountersModule();
+        }
+        catch (Exception e) {
+            throw new IllegalStateException("Could not initialize Jvm Performance Counters module "
+                + "please set the property 'azure.application-insights.default.modules.JvmPerformanceCountersModule.enabled=false' to "
+                + "avoid this error message", e);
+        }
+    }
+
+    @Bean
+    @ConditionalOnProperty(value = "azure.application-insights.default.modules.HearBeat.enabled", havingValue = "true", matchIfMissing = true)
```

Comment:
Typo `HearBeat` to `HeartBeat`.
Also isn't it web module? If so it's better to move it under web auto-configuration.

Reply:
@gavlyukovskiy yes I will fix this typo. No it is not web module. It is core module and is expected to work even in worker role scenarios. I think this is a right place for it.

Reply:
Done